src/fullnode.rs

```diff
@@ -108,17 +107,16 @@ pub struct Fullnode {
     rpc_addr: SocketAddr,
     rpc_pubsub_addr: SocketAddr,
     drone_addr: SocketAddr,
-    vote_signer_addr: SocketAddr,
     db_ledger: Arc<DbLedger>,
+    vote_signer: Arc<VoteSignerProxy>,
 }
 
 impl Fullnode {
     pub fn new(
         node: Node,
         ledger_path: &str,
         keypair: Arc<Keypair>,
-        vote_account_id: &Pubkey,
-        vote_signer_addr: &SocketAddr,
+        vote_signer: Arc<VoteSignerProxy>,
```

Comment:
Doesn't seem like that `Arc` should be needed yet. Is the same object shared across the fullnode and some other thread that's *not* created by fullnode?

Reply:
If actually shared, then yes, you'll need the Arc. If it can be passed from fullnode to replay_stage on construction and then only referenced by replay_stage, then you don't need it.

Reply:
It's the former. The object is stored in fullnode, and also passed on to replay stage. It's stored in fullnode for leader <-> validator role transition.